src/group_theory/group_action/defs.lean

```diff
@@ -191,21 +195,22 @@ variables {M}
 /-- Note that the `smul_comm_class M α α` typeclass argument is usually satisfied by `algebra M α`.
 -/
 @[to_additive]
-lemma mul_smul_comm [monoid α] (s : M) (x y : α) [smul_comm_class M α α] :
+lemma mul_smul_comm [has_mul α] (s : M) (x y : α) [smul_comm_class M α α] :
   x * (s • y) = s • (x * y) :=
 (smul_comm s x y).symm
 
 /-- Note that the `is_scalar_tower M α α` typeclass argument is usually satisfied by `algebra M α`.
 -/
-lemma smul_mul_assoc [monoid α] (r : M) (x y : α) [is_scalar_tower M α α] :
+lemma smul_mul_assoc [has_mul α] (r : M) (x y : α) [is_scalar_tower M α α] :
   (r • x) * y = r • (x * y) :=
 smul_assoc r x y
 
 /-- Note that the `is_scalar_tower M α α` and `smul_comm_class M α α` typeclass arguments are
 usually satisfied by `algebra M α`. -/
-lemma smul_mul_smul [monoid α] (r s : M) (x y : α) [is_scalar_tower M α α] [smul_comm_class M α α] :
+lemma smul_mul_smul [has_mul α] (r s : M) (x y : α)
+  [is_scalar_tower M α α] [smul_comm_class M α α] :
   (r • x) * (s • y) = (r * s) • (x * y) :=
-by rw [smul_mul_assoc, mul_smul_comm, smul_smul]
+by rw [smul_mul_assoc, mul_smul_comm, ← smul_assoc, smul_eq_mul]
```

Comment:
If I'm reading this correctly, then the two typeclasses above actually imply `mul_smul`? I guess they don't imply `one_smul` though so `mul_action` is still useful

Reply:
If I understand you correctly, you're observing the following works:
```lean
import group_theory.group_action.defs

example (α β : Type*) [monoid α] [has_scalar α β] [is_scalar_tower α α β] : mul_action α β :=
{ one_smul  := sorry,
  mul_smul := λ x y b, by rw [← smul_eq_mul, smul_assoc], }
```

Reply:
Exactly - but I don't think there's any sensible way to act on that observation.